b1ddi/resource_ipam_subnet.go

```diff
@@ -340,54 +345,82 @@ func resourceIpamsvcSubnet() *schema.Resource {
 
 func resourceIpamsvcSubnetCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	c := m.(*b1ddiclient.Client)
+	var id, address string
+
+	address = d.Get("address").(string)
+
+	if strings.HasPrefix(address, "ipam/address_block") {
```

Comment:
Shouldn't we have used the existing `parent` field for this logic?

Reply:
How do you see using the `parent` field help here? Regardless of the subnet being a static assignment or dynamically created through the Next available API, the `parent` would be either an Address Block or an IP Space.

With the `address` field, we would be able to distinguish the static vs dynamic easily

Reply:
`address` is a required field. If we rely on `parent` for the NASub case, what will be set in the `address`?

Reply:
Also, `parent` is a computed field.

Reply:
You can keep it optional and have `ExactlyOneOf` in the schema to allow either `parent` or `address` and not both

Reply:
Same with `parent`. It is computed because it is read only to the API. You can certainly change it to optional in the provider, and not send the value to the API. The parent field will only be used to trigger the NAS call.

Reply:
> You can keep it optional and have `ExactlyOneOf` in the schema to allow either `parent` or `address` and not both

I'll try this out